src/test_application/flight.cpp

```diff
@@ -0,0 +1,167 @@
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include "flight.h"
+
+#include "DebugObject.h"
+
+#include <osp/Active/ActiveScene.h>
+#include <osp/Active/SysDebugRender.h>
+#include <osp/Active/SysVehicle.h>
+#include <osp/Active/SysForceFields.h>
+#include <osp/Active/SysAreaAssociate.h>
+
+#include <adera/Machines/UserControl.h>
+#include <adera/Machines/Rocket.h>
+
+#include <planet-a/Active/SysPlanetA.h>
+#include <planet-a/Satellites/SatPlanet.h>
+
+using osp::Vector2;
+
+using osp::Vector3;
+using osp::Quaternion;
+using osp::Matrix3;
+using osp::Matrix4;
+
+// for the 0xrrggbb_rgbf and angle literals
+using namespace Magnum::Math::Literals;
+
+using osp::universe::Universe;
+using osp::universe::Satellite;
+using osp::universe::SatActiveArea;
+
+using osp::universe::UCompActiveArea;
+
+using osp::active::ActiveEnt;
+using osp::active::ACompTransform;
+using osp::active::ACompCamera;
+using osp::active::ACompFloatingOrigin;
+
+using adera::active::machines::SysMachineUserControl;
+using adera::active::machines::SysMachineRocket;
+
+using planeta::universe::SatPlanet;
+
+
+void test_flight(std::unique_ptr<OSPMagnum>& magnumApp,
+                 osp::OSPApplication& ospApp, OSPMagnum::Arguments args)
+{
+
+    // Get needed variables
+    Universe &uni = ospApp.get_universe();
+    SatActiveArea *satAA = static_cast<osp::universe::SatActiveArea*>(
+            uni.sat_type_find("Vehicle")->second.get());
+    SatPlanet *satPlanet = static_cast<planeta::universe::SatPlanet*>(
+            uni.sat_type_find("Planet")->second.get());
+
+    // Create the application
+    magnumApp = std::make_unique<OSPMagnum>(args, ospApp);
+
+    // Configure the controls
+    config_controls(*magnumApp);
+
+    // Create an ActiveScene
+    osp::active::ActiveScene& scene = magnumApp->scene_add("Area 1");
+
+    // Register dynamic systems for that scene
+
+    auto &sysPhysics = scene.dynamic_system_add<osp::active::SysPhysics>(
+                "Physics");
```

Comment:
I'm wary of raw strings like this. Strings are expensive to do comparisons.

Could you make the strings be static global variables so that a comparison of the pointer values is sufficient for unique identification?

Reply:
These functions aren't performance critical; there's not much to worry about. This name can be stored as a static constant in SysPhysics and other system classes, similar to the Sat* classes mentioned in the previous comment. The name wouldn't be needed by the function at all:
```cpp
auto &sysPhysics = scene.dynamic_system_add<osp::active::SysPhysics>();
```